docs/checks/reader.go

```diff
@@ -16,36 +16,70 @@
 package checks
 
 import (
-
-	// Used to embed `checks.yaml` file.
-	_ "embed"
+	"errors"
 	"fmt"
+	"strings"
+
+	"internal/idocs"
 
-	"gopkg.in/yaml.v2"
+	sce "github.com/ossf/scorecard/v2/errors"
 )
 
-//go:embed checks.yaml
-var checksYAML []byte
+var errCheckNotExist = errors.New("check does not exist")
 
-// Check defines expected check definition in checks.yaml.
-type Check struct {
-	Risk        string   `yaml:"-"`
-	Short       string   `yaml:"short"`
-	Description string   `yaml:"description"`
-	Tags        string   `yaml:"tags"`
-	Remediation []string `yaml:"remediation"`
+const docURL = "https://github.com/ossf/scorecard/blob/main/docs/checks.md"
+
+// DocInterface defines the documentation interface.
+type DocInterface interface {
```

Comment:
Can we avoid this interface? IMO, it's an overkill for this usecase. Why not move all the doc generation code to an `internal` dir and package? That way we still do not expose whats not needed?

Reply:
I guess having this interface allows us to import this package in `pkg`. But don't we end up exposing `DocInterface` now? How is that better than what we have today?

Reply:
it's useful for the unit tests. Initially I did not have it and I could not create a `Doc` for the unit tests so I had to create one :/
let me know if you have a better fix. It's the first that came to mind

Reply:
I see, in that case:

i) Rename the file to `doc.go` or some equivalent. 
ii) Remove the `Interface` suffix.
ii) Move `type Doc struct` also inside internal or maybe a different file to separate out the interface and its implementation.

Reply:
done. I created a `doc.go` that holds the interfaces; and a `impl.go` that holds the implementation. If we ever find a way to remove the interface, we'll simply rename the implementation to the interface name and nothing should break.